packages/flutter/test/services/delta_text_input_test.dart

```diff
@@ -65,6 +66,162 @@ void main() {
         expect(client.latestMethodCall, 'updateEditingValueWithDeltas');
       },
     );
+
+    test('Invalid TextRange fails loudly when being converted to JSON - NonTextUpdate', () async {
+      final List<FlutterErrorDetails> record = <FlutterErrorDetails>[];
+      FlutterError.onError = (FlutterErrorDetails details) {
+        record.add(details);
+      };
+
+      final FakeDeltaTextInputClient client = FakeDeltaTextInputClient(const TextEditingValue(text: '1'));
+      const TextInputConfiguration configuration = TextInputConfiguration(enableDeltaModel: true);
+      TextInput.attach(client, configuration);
+
+      const String jsonDelta = '{'
+          '"oldText": "1",'
+          ' "deltaText": "",'
+          ' "deltaStart": -1,'
+          ' "deltaEnd": -1,'
+          ' "selectionBase": 3,'
+          ' "selectionExtent": 3,'
+          ' "selectionAffinity" : "TextAffinity.downstream" ,'
+          ' "selectionIsDirectional": false,'
+          ' "composingBase": -1,'
+          ' "composingExtent": -1}';
+
+      final ByteData? messageBytes = const JSONMessageCodec().encodeMessage(<String, dynamic>{
+        'method': 'TextInputClient.updateEditingStateWithDeltas',
+        'args': <dynamic>[-1, jsonDecode('{"deltas": [$jsonDelta]}')],
+      });
+
+      await ServicesBinding.instance.defaultBinaryMessenger.handlePlatformMessage(
+        'flutter/textinput',
+        messageBytes,
+        (ByteData? _) {},
+      );
+      expect(record.length, 1);
+      // Verify the error message in parts because Web formats the message
+      // differently from others.
+      expect(record[0].exception.toString(), matches(RegExp(r'\brange.start >= 0 && range.start <= text.length\b')));
+      expect(record[0].exception.toString(), matches(RegExp(r'\bRange start 3 is out of text of length 1\b')));
```

Comment:
out of curiosity, what's the difference?

Reply:
Slight formatting difference.

```
    Actual: '\'package:flutter/src/services/text_editing_delta.dart\': Failed assertion: line 140 pos 14: \'_debugTextRangeIsValid(newSelection, oldText)\': The selection range: TextSelection.collapsed(offset: 3, affinity: TextAffinity.downstream, isDirectional: false) is not within the bounds of text: 1 of length: 1'
```

```
    Actual: 'Assertion failed: file:///Users/roliv/flutter/packages/flutter/lib/src/services/text_editing_delta.dart:140:14\n'
              '_debugTextRangeIsValid(newSelection, oldText)\n'
              '"The selection range: TextSelection.collapsed(offset: 3, affinity: TextAffinity.downstream, isDirectional: false) is not within the bounds of text: 1 of length: 1"'
```